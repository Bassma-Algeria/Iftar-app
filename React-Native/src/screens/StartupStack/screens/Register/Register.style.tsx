import {Dimensions, StyleSheet} from 'react-native';

const windowHeight = Dimensions.get('window').height;

const styles = StyleSheet.create({
  container: {
    marginTop: windowHeight <= 700 ? -100 : -90,
  },

  form: {
    marginTop: 20,
  },

  buttonContainer: {
    marginTop: 30,
  },

  spaceInputs: {
    marginTop: 20,
    marginBottom: 10,
  },
  text: {
    textDecorationLine: 'underline',
  },
  registerButton: {
    marginTop: 60,
  },
});

export {styles};
