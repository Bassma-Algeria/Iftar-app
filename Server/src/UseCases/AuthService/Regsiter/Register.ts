export interface RegisterBody {
  email: string;
  password: string;
  phoneNumber: string;
}

class RegisterFactory {
  constructor() {}

  async register(userInfo: RegisterBody) {
    
  }
}

export { RegisterFactory };
