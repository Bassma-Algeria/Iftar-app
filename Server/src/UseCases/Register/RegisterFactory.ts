import type { IRestaurantOwnersGateway } from "../../Ports/DrivenPorts/Persistence/RestaurantOwnersGateway/RestaurantOwnersGateway.interface";
import type { IPasswordManager } from "../../Ports/DrivenPorts/PasswordManager/PasswordManager.interface";

import { RestaurantOwner } from "../../Domain/RestaurantOwner/RestaurantOwner";
import type { IRestaurantOwner } from "../../Domain/RestaurantOwner/RestaurantOwnerFactory";

interface RegistrationBody {
  email: string;
  password: string;
  confirmPassword: string;
  phoneNumber: string;
}

class RegisterFactory {
  constructor(
    private readonly restaurantOwnersGateway: IRestaurantOwnersGateway,
    private readonly passwordManager: IPasswordManager
  ) {}

  async register(registrationBody: RegistrationBody) {
    const owner = new RestaurantOwner(registrationBody);

    const confirmPassword = registrationBody.confirmPassword.trim();
    if (owner.password !== confirmPassword) this.WrongConfirmPasswordException();

    const ownerWithSameEmail = await this.findOwnerByEmail(owner.email);
    if (ownerWithSameEmail) this.EmailExistException();

    const ownerWithSamePhone = await this.findOwnerByPhone(owner.phoneNumber);
    if (ownerWithSamePhone) this.PhoneExistException();

    owner.password = await this.passwordManager.hash(owner.password);

    await this.saveOwner(owner);
  }

  private async saveOwner(owner: IRestaurantOwner) {
    await this.restaurantOwnersGateway.save(owner);
  }

  private findOwnerByEmail(email: string) {
    return this.restaurantOwnersGateway.getByEmail(email);
  }

  private findOwnerByPhone(phoneNumber: string) {
    return this.restaurantOwnersGateway.getByPhone(phoneNumber);
  }

  private WrongConfirmPasswordException(): never {
    throw { confirmPassword: "Confirm Password didn't match the password" };
  }

  private EmailExistException(): never {
    throw { email: "email already used" };
  }

  private PhoneExistException(): never {
    throw { phoneNumber: "phone number already used" };
  }
}

export { RegisterFactory };
