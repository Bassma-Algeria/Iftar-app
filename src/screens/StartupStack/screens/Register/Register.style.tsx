import {Dimensions, StyleSheet} from 'react-native';

const windowHeight = Dimensions.get('window').height;

const styles = StyleSheet.create({
  container: {
    marginTop: windowHeight <= 850 ? -110 : -60,
  },

  form: {
    marginTop: 20,
  },

  buttonContainer: {
    marginTop: 30,
  },

  spaceInputs: {
    marginBottom: 22,
  },

  text: {
    textDecorationLine: 'underline',
  },

  registerButton: {
    marginTop: 60,
  },
});

export {styles};
