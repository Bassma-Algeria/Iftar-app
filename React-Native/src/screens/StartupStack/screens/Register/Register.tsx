import React, {useState} from 'react';
import {View} from 'react-native';

import {styles} from './Register.style';

import {ICONS} from '../../../../utils/constants/Icons';

import {Layout} from '../../_components_/Layout/Layout';
import {Header} from '../../../../components/Header/Header';
import {Input} from '../../../../components/Inputs/TextInput/Input';
import {PasswordInput} from '../../../../components/Inputs/passwordInput/PasswordInput';
import {Button} from '../../../../components/Button/Button';

import {SignupInfo} from '../../../../Gateways/AuthGateway/AuthGateway.interface';
import {RestaurantsOwnersFakeGateway} from '../../../../Gateways/AuthGateway/Fake/RestaurantOwnersFakeGateway';

interface Props {}
interface FormInfo {
  email: string;
  password: string;
  confirmPassword: string;
  phoneNumber: string;
}

const Register: React.FC<Props> = () => {
  const [loading, setLoading] = useState(false);
  const [formInfo, setformInfo] = useState<FormInfo>({
    email: '',
    password: '',
    confirmPassword: '',
    phoneNumber: '',
  });
  const register = (userInfo: SignupInfo) => {
    const RestaurentOwner = new RestaurantsOwnersFakeGateway();
    if (!(userInfo.email || userInfo.password)) {
    } else {
      setLoading(true);
      RestaurentOwner.login(userInfo)
        .then((resp: any) => {
          console.log(resp);
        })
        .catch(err => {
          setCredentialsError(err.message);
        })
        .finally(() => {
          setLoading(false);
        });
    }
  };
  return (
    <Layout style={styles.container}>
      <Header color="brown" align="center" variant="h1" fontWeight="semibold">
        تسجيل
      </Header>
      <RegisterationForm />
      <View style={styles.buttonContainer}>
        <Button label="سجل الآن" />
      </View>
    </Layout>
  );
};

interface RegisterProps {}
const RegisterationForm: React.FC<RegisterProps> = () => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');
  const [phone, setPhone] = useState<string>('');
  return (
    <>
      <Input
        text={email}
        setText={setEmail}
        label="البريد الالكتروني"
        keyboardType="default"
        icon={ICONS.email}
        style={styles.spaceInputs}
      />
      <PasswordInput
        password={password}
        setPassword={setPassword}
        label="كلمة المرور"
        style={styles.spaceInputs}
      />
      <PasswordInput
        password={confirmPassword}
        setPassword={setConfirmPassword}
        label="تأكيد كلمة المرور"
        style={styles.spaceInputs}
      />

      <Input
        text={phone}
        setText={setPhone}
        label="رقم الهاتف"
        keyboardType="numeric"
        icon={ICONS.phoneLightColor}
        style={styles.spaceInputs}
      />
    </>
  );
};
export {Register};
