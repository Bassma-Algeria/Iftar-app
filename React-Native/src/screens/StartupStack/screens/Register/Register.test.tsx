import React from 'react';
import renderer from 'react-test-renderer';

import {Register} from './Register';

describe('<RegisterScreen />', () => {
  it('should render correctly', () => {
    const props: any = {};
    const tree = renderer.create(<Register {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
