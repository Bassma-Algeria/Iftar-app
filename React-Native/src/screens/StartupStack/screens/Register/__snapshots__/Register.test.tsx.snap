// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen /> should render correctly 1`] = `<View />`;
